packages/algob/src/lib/deployer.ts

```diff
@@ -85,13 +89,14 @@ export class AlgobDeployerImpl implements AlgobDeployer {
     return this.cpData.precedingCP[this.networkName].asa[name];
   }
 
-  async deployASC (name: string, source: string, account: Account): Promise<ASCInfo> {
+  async deployASC (name: string, scParams: Object, flags: ASCDeploymentFlags): Promise<ASCInfo> {
+    const result: ASCCache = await this.algoOp.ensuredCompiled(name, false);
```

Comment:
You compile the file before you check that the code is deployed (`assertNoAsset`). Why would you recompile when it's deployed?
Move this into algoOp and remove `ensuredCompiled` from `algoOp`

Reply:
this is exactly what this code is doing.

Reply:
BTW: the `force` argument should be provided by user in a command line (don't assume it's `false`)

Reply:
I moved this part to algoOp, that is after assertNoAsset